### DISCUSSION TITLE
Sharing Slash command parent Group between Cogs

### DISCUSSION THREAD

**Discussion by MebinAbraham** [Rapptz/discord.py#8069] [Q&A]
The current library design doesn't seem to allow you to have sub commands for a parent command in multiple cogs. This makes it really hard to split out the code when you have large number of sub commands, especially if you have nested sub commands. Is this currently possible. For example I am looking get something like this working.

```
/parent
   child-1 (cog)
       cmd-1
       cmd-2
       ...
   child-2 (cog)
       cmd-1
       cmd-2
       ...
```

Is it possible to get this pattern working with the current lib?. I'm sure it is possible with some messing about, but wondering if there was a nice way to achieve it instead of a hacky way. When I inherit or refer class groups, it doesn't work. If I were to make a group in a non-cog file and import, then commands can sync but an exception would be raised due to argument mismatch as it doesn't pass self.

It does work in another library which I am migrating from. It isn't flexible to put all sub-command and nested sub-commands into 1 cog. Any ideas/suggestions?. Thanks

**Comment by Rapptz** [comment]
There is unfortunately no plan to add support for this due to how complicated bindings become when they escape their original extension. It makes ownership of extensions quite complicated and is not something I want to mess with at this time.

**Reply by MebinAbraham** [reply]
No probs, thanks for the update.
